src/Adapter/Order/CommandHandler/AddCartRuleToOrderHandler.php

```diff
@@ -69,6 +76,10 @@ public function handle(AddCartRuleToOrderCommand $command): void
         $this->assertPercentCartRule($command);
         $order = $this->getOrder($command->getOrderId());
 
+        $this->contextStateManager
```

Comment:
You need to `restoreContext` in this handler using a `try/finally` that encapsulates everything after this call (maybe use a function to avoid the diff to become unreadable just because of the indentation